client/packages/programs/src/JsonForms/common/components/Select.tsx

```diff
@@ -168,22 +185,69 @@ const getHighlightParts = (
   );
 };
 
+/** Either the full list or the narrowed list from conditional option */
+
+const useFilteredItems = (
+  items: string[] | undefined,
+  current: string | undefined,
```

Comment:
Maybe these variables could be called `allItems` and `currentItems` - it makes it clearer that one is a subset of the other.

Reply:
thanks I called it currentSelection and added some more in the comment